### ISSUE TITLE
Ability to remove, remap default Menu mappings

### ISSUE THREAD

**Issue by shemerey** [rogual/neovim-dot-app#45]
It would be nice to have something like `macmenu` for `MacVim` 

For instance if you would like, to remove or remap default menu mapping in `MacVim` you can do like so: 

```
macmenu File.Print key=<nop>
```

**Comment by iansinnott** [comment]
:+1: This is a great feature of MacVim. It would be nice to see it supported here as well

**Comment by rogual** [comment]
Some thoughts on how the API for this might look:

**Add/remove**
call MacMenu(["File", "Print"], "C-p", PrintSomething)
call MacMenu(["File", "Print"], PrintSomething)
call MacMenuRemove(["File", "Print"])

**Before/after**
call MacMenuAfter(["File", "Save Buffer"], "Save Buffer Seriously", DoIt)
call MacMenuSeparatorAfter(["File", "Print"])
call MacMenuRemoveSeparatorAfter(["File", "Print"])

**Get reference**
asmMenu = MacMenu(["Asm"])     -- create if not exist
asmMenu = MacMenuFind(["Asm"]) -- return null if not exist

**Query**
MacMenuGetCaption(asmMenu)
MacMenuGetShortcut(asmMenu)
MacMenuGetAction(asmMenu)

**Alter**
MacMenuSetCaption(asmMenu, "Spoon")
MacMenuSetShortcut(asmMenu, "C-s")
MacMenuSetAction(asmMenu, DoStuff)

**Tree**
MacMenuGetParent(asmMenu)
MacMenuGetChildren(asmMenu)

**Referencing menus**
Strings: ["File", "Print"]
Refs: printMenu
Combo: [fileMenu, "Print"]

**Comment by rogual** [comment]
Actually, since Neovim hasn't removed Vim's own menu system, let's use that! Then we'll only need a couple of commands to set Mac key equivalents.

Started work on this in the `menu` branch.

**Comment by rogual** [comment]
Implemented in this PR: https://github.com/rogual/neovim-dot-app/pull/101

This needed a couple of changes to Neovim itself, so we can't merge until they accept those.

**Comment by shemerey** [comment]
awesome work, looking forward to play with it

**Comment by rogual** [comment]
https://github.com/rogual/neovim-dot-app/pull/101 merged, have fun, beware bugs
